packages/@aws-cdk/aws-sagemaker/lib/endpoint-config.ts

```diff
@@ -0,0 +1,313 @@
+import { EOL } from 'os';
+import * as kms from '@aws-cdk/aws-kms';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { AcceleratorType } from './accelerator-type';
+import { InstanceType } from './instance-type';
+import { IModel } from './model';
+import { sameEnvDimension } from './private/util';
+import { CfnEndpointConfig } from './sagemaker.generated';
+
+/**
+ * The interface for a SageMaker EndpointConfig resource.
+ */
+export interface IEndpointConfig extends cdk.IResource {
+  /**
+   * The ARN of the endpoint configuration.
+   *
+   * @attribute
+   */
+  readonly endpointConfigArn: string;
+  /**
+   * The name of the endpoint configuration.
+   *
+   * @attribute
+   */
+  readonly endpointConfigName: string;
+}
+
+/**
+ * Common construction properties for all production variant types (e.g., instance, serverless).
+ */
+interface ProductionVariantProps {
+  /**
+   * Determines initial traffic distribution among all of the models that you specify in the
+   * endpoint configuration. The traffic to a production variant is determined by the ratio of the
+   * variant weight to the sum of all variant weight values across all production variants.
+   *
+   * @default 1.0
+   */
+  readonly initialVariantWeight?: number;
+  /**
+   * The model to host.
+   */
+  readonly model: IModel;
+  /**
+   * Name of the production variant.
+   */
+  readonly variantName: string;
+}
+
+/**
+ * Construction properties for an instance production variant.
+ */
+export interface InstanceProductionVariantProps extends ProductionVariantProps {
+  /**
+   * The size of the Elastic Inference (EI) instance to use for the production variant. EI instances
+   * provide on-demand GPU computing for inference.
+   *
+   * @default - none
+   */
+  readonly acceleratorType?: AcceleratorType;
+  /**
+   * Number of instances to launch initially.
+   *
+   * @default 1
+   */
+  readonly initialInstanceCount?: number;
+  /**
+   * Instance type of the production variant.
+   *
+   * @default - ml.t2.medium instance type.
```

Comment:
```suggestion
   * @default - InstanceType.T2_MEDIUM
```

or whatever its actually called

Reply:
Is it safe to assume the ` - ` can be left off since `InstanceType.T2_MEDIUM` is referencing a real value?